homeassistant/components/esphome/__init__.py

```diff
@@ -114,8 +114,17 @@ def async_on_service_call(service: ServiceCall) -> None:
                 _LOGGER.error('Error rendering data template: %s', ex)
                 return
 
-        hass.async_create_task(hass.services.async_call(
-            domain, service_name, service_data, blocking=True))
+        if service.is_event:
+            # ESPHome uses servicecall packet for both events and service calls
+            # Ensure the user can only send events of form 'esphome.xyz'
+            if domain != 'esphome':
+                raise ValueError("Can only generate events under esphome "
```

Comment:
Where is this caught? Integrations should normally avoid leaking exceptions. We usually log error and return.

Reply:
The error is caught by HA - it will result in an error in the HA logs and no information leakage.

I can make it an error message too though.